### ISSUE TITLE
Make ODataConversionModelBuilder accept an existing model

### ISSUE THREAD

**Issue by chinadragon0515** [OData/WebApi#755]
Make ODataConversionModelBuilder accept an existing model

In RESRier, there is always a model built by default, user need to extend the model, and we need a better way for user to extend the model.

And to have it in breaking changes.
### Additional details

Per discussion, make ODataConversionModelBuilder to have a new construct to accept an existing model.

**Comment by xuzhg** [comment]
Can model reference work for your scenario?

```c#
// existing model
EdmModel existingModel = ....;

// build new model:
var builder = new ODataConventionModelBuilder();
....
IEdmModel model = buider.GetEdmModel();

// add model reference
existingModel .AddReferencedModel(model);
```

**Comment by chinadragon0515** [comment]
I tried with model references and also did some test,

1.	For the ODL model reference example, it is based on CsdlSemanticsModel (Super class is EdmModelBase),  it already has CSDL files, when create a model, it will read content from CSDL files includes the “Includes” for model reference
When it writes the model, it will write the Includes in model reference of the main model, this is the reason we see some reference in the $metadata

2.	With OData conversion model builder, EdmModel is used (Super class is EdmModelBase),  I did not see “Includes” related logic.

3.	I did not see any logic to write ReferenceModels related information except includes I mentioned above.

4.	About the code usage of RerefenceModels, here are some usages, mainly used by validation for duplicate items check and found operation/entity type (not entity set) by ODataUriResolver.

5.	Test been done,
a.	Define duplicate items in reference mode, and call model validation, get errors output as expected (code logic will continue)
b.	Define entity set in reference model and request that entity set, URI parsing will fail and cannot find the resource.
c.	Define function import in reference model, and do request, cannot parse the URL

Issues:
1.	Reference models for EdmModel has no reference in $metatada.
2.	Content in reference model is not used to parse URL.
